lib/ecto/exceptions.ex

```diff
@@ -181,7 +181,49 @@ defmodule Ecto.NoPrimaryKeyValueError do
 end
 
 defmodule Ecto.ChangeError do
-  defexception [:message]
+  defexception [:action, :schema, :field, :type, :value]
+
+  def exception(opts) do
+    struct!(__MODULE__, opts)
+  end
+
+  def message(%__MODULE__{type: :decimal, value: %Decimal{coef: coef}} = exception) when coef in [:inf, :qNaN, :sNaN] do
+    common_message(exception) <>
+      """
+
+
```

Comment:
need 2x \n so it looks like this:

```
** (Ecto.ChangeError) value `#Decimal<Infinity>` for `Post.d` in `insert` does not match type :decimal

`+Infinity`, `-Infinity`, and `NaN` values are not supported, even though the `Decimal` library handles them.
To support them, you can create a custom type.
```

Reply:
So let's use `\n\n` like the other ones? Also, we should end the next line with a `\` so we don't force a new line on users. Or we use regular strings instead of heredocs.

Reply:
Minor nitpick: should backticks be removed from the message?

```
** (Ecto.ChangeError) value #Decimal<Infinity> for Post.d in "insert" does not match type :decimal

+Infinity, -Infinity, and NaN values are not supported, even though the Decimal library handles them.
To support them, you can create a custom type.
```